src/Symfony/Component/DependencyInjection/Compiler/ReplaceAliasByActualDefinitionPass.php

```diff
@@ -109,12 +109,12 @@ private function updateReferences($container, $currentId, $newId)
      */
     private function updateArgumentReferences(array $arguments, $currentId, $newId)
     {
-        foreach ($arguments as $k => $argument) {
+        foreach ($arguments as $k => &$argument) {
```

Comment:
:-1: for using refs here and below: it leaves the var as ref outside of the loop, which is error prone. It also favors mem copies because of the way php works internally and often led to perf pb.